### PR TITLE
Rename AsyncRandom and nextPredictable

### LINKED ISSUE #3180: Missing verb in the documentation for the Xoroshiro128PlusPlusRandom#next method
https://github.com/FabricMC/yarn/blob/2f8ce2e639fdfc6f8d873773c21e5cb130516607/mappings/net/minecraft/util/math/random/Xoroshiro128PlusPlusRandom.mapping#L17-L19

### PR DESCRIPTION
**Pull request by apple502j**
See #3140 reviews
Fixes #3180

### CODE REVIEW

File: mappings/net/minecraft/util/math/random/Random.mapping [hunk 1]
```diff
@@ -54,6 +54,6 @@ CLASS net/minecraft/class_5819 net/minecraft/util/math/random/Random
 	METHOD method_43057 nextFloat ()F
 	METHOD method_43058 nextDouble ()D
 	METHOD method_43059 nextGaussian ()D
-	METHOD method_43385 nextPredictable (DD)D
+	METHOD method_43385 nextTriangular (DD)D
 		ARG 1 base
 		ARG 3 variance
```
Comment (liach) [inline mappings/net/minecraft/util/math/random/Random.mapping]: This isn't a variance, but more like a "radius", like the difference from the mean (should rename "base" too) to the upper and lower limit.

Comment (haykam821) [reply mappings/net/minecraft/util/math/random/Random.mapping]: Would `deviation` work?

```suggestion
		ARG 3 deviation
```

Comment (liach) [reply mappings/net/minecraft/util/math/random/Random.mapping]: > Would `deviation` work?

Sure, as long as it doesn't overlap with statistical terms.

**Comment by liach** [comment]
Also remember to update the reference to `AsyncRandom` in `Random` interface documentation

File: mappings/net/minecraft/util/math/random/Random.mapping (hunk 1 above)
Comment (haykam821) [reply mappings/net/minecraft/util/math/random/Random.mapping]: These are the typical statistical terms when describing a triangular distribution:

triangular-distribution

The parameters `mean` and `deviation` would align with these terms.

**Comment by apple502j** [comment]
Note that for triangular distribution the peak is the mode. While in this case it equals the mean we should use commonly accepted terms.

File: mappings/net/minecraft/util/math/random/Random.mapping [hunk 2]
```diff
@@ -54,6 +54,8 @@ CLASS net/minecraft/class_5819 net/minecraft/util/math/random/Random
 	METHOD method_43057 nextFloat ()F
 	METHOD method_43058 nextDouble ()D
 	METHOD method_43059 nextGaussian ()D
-	METHOD method_43385 nextPredictable (DD)D
-		ARG 1 base
-		ARG 3 variance
+	METHOD method_43385 nextTriangular (DD)D
+		COMMENT {@return a random {@cpde double} between {@code mode - deviation} and
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/random/Random.mapping]:
```suggestion
		COMMENT {@return a random {@code double} between {@code mode - deviation} and
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/util/math/random/Random.mapping [hunk 3]
```diff
@@ -54,6 +54,8 @@ CLASS net/minecraft/class_5819 net/minecraft/util/math/random/Random
 	METHOD method_43057 nextFloat ()F
 	METHOD method_43058 nextDouble ()D
 	METHOD method_43059 nextGaussian ()D
-	METHOD method_43385 nextPredictable (DD)D
-		ARG 1 base
-		ARG 3 variance
+	METHOD method_43385 nextTriangular (DD)D
+		COMMENT {@return a random {@code double} between {@code mode - deviation} and
+		COMMENT {@code mode + deviation} (both inclusive) with mode {@code mode}}
+		ARG 1 mode
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/random/Random.mapping]: The `mode` parameter should be named `mean` to align with other Yarn parameter names. For example, the `MathHelper.nextGaussian` method has `mean` and `deviation` parameters.

```suggestion
		COMMENT {@return a random {@code double} between {@code mean - deviation} and
		COMMENT {@code mean + deviation} (both inclusive) with mean {@code mean}}
		ARG 1 mean
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/math/random/Random.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/util/math/random/Random.mapping]: It's simultaneously mode and mean.

Comment (haykam821) [reply mappings/net/minecraft/util/math/random/Random.mapping]: I think that Yarn should standardize on using mean.

Comment (liach) [reply mappings/net/minecraft/util/math/random/Random.mapping]: Or just `center` as this is the center of the triangular distribution, more straightforward to general users

Comment (apple502j) [reply mappings/net/minecraft/util/math/random/Random.mapping]: Triangular distribution in general does not have to be symmetric, and in that case the peak is not the mean. Here both works but this is fine.

Gaussian distribution's μ is the mean - never heard it as the mode, even though mathematically it is the same.

**Review by haykam821 — APPROVED** [review]

**Comment by liach** [comment]
> Triangular distribution in general does not have to be symmetric, and in that case the peak is not the mean. Here both works but this is fine.

Then I recommend you document that this triangular distribution is symmetric, and thus its mean, median, and mode is the same value `mode` passed.

**Review by haykam821 — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**